Formula/php@7.2.rb

```diff
@@ -19,7 +19,8 @@ class PhpAT72 < Formula
 
   keg_only :versioned_formula
 
-  deprecate! date: "2021-11-30", because: :deprecated_upstream
+  # See: https://www.php.net/eol.php
+  deprecate! date: "2020-11-30", because: :deprecated_upstream
```

Comment:
Why not just `disable!` it for `2021-11-30`? That's the same as doing `deprecate!` now and then changing it to `disable!` at the end of November, except with less work.

Reply:
That's how this was originally handled and I agree it makes sense. I'll update this accordingly.

This was switched from `disable!` to `deprecate!` in #77283 without modifying the date, which essentially delayed the deprecation. I don't agree with that approach because PHP 7.2 is unsupported as of 2020-11-30 and folks should know about it now (not a year later), so they migrate to a supported version.

The change in that PR was intended to work around the fact that deprecated formulae are skipped in tests, so `php@7.2` wasn't revision bumped when it should have been. In my opinion, this was a hacky way of avoiding having to deal with a more complex issue (deprecated formulae that may need a revision bump).

Reply:
Can we leave a comment here that states the future `disable!` date is intentional, and that it shouldn't be changed?

Reply:
Sure, better safe than sorry. Do you think the comment in the latest push is sufficient?

Reply:
yep, looks good